subid: drop support for old version

giuseppe opened the pull request:
the new ABI has been around for more than 3 years now (shadow-utils v4.11).

Closes: https://github.com/containers/storage/issues/2260

Luap99 reviewed (changes requested):
This breaks build support for RHEL 9
```
[root@vm-10-0-187-181 ~]# ls /usr/lib64/libsubid.so.3
/usr/lib64/libsubid.so.3
[root@vm-10-0-187-181 ~]# ls /usr/lib64/libsubid.so.4
ls: cannot access '/usr/lib64/libsubid.so.4': No such file or directory
[root@vm-10-0-187-181 ~]# cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="9.6 (Plow)"
...
```

I don't think we can drop support for RHEL 9 at this point?

giuseppe commented:
> This breaks build support for RHEL 9
> 
> ```
> [root@vm-10-0-187-181 ~]# ls /usr/lib64/libsubid.so.3
> /usr/lib64/libsubid.so.3
> [root@vm-10-0-187-181 ~]# ls /usr/lib64/libsubid.so.4
> ls: cannot access '/usr/lib64/libsubid.so.4': No such file or directory
> [root@vm-10-0-187-181 ~]# cat /etc/os-release 
> NAME="Red Hat Enterprise Linux"
> VERSION="9.6 (Plow)"
> ...
> ```
> 
> I don't think we can drop support for RHEL 9 at this point?

shurg, I assumed we moved on from that old version.  Thanks for checking it.  I've fixed the patch

Luap99 reviewed (commented):
Now that I look at it I remembered https://issues.redhat.com/browse/RHEL-21366.
I think this would fix that as we correctly use the init function now?

Luap99 reviewed (approved):
LGTM I suppose but I have not tried to compile test it

giuseppe commented:
@kolyshkin PTAL

giuseppe commented:
@mtrmac PTAL

giuseppe commented:
@flouthoc PTAL

flouthoc reviewed (approved):
LGTM

flouthoc commented:
/lgtm
/approve
